packages/clay-alert/src/ClayAlertBase.js

```diff
@@ -113,11 +101,30 @@ class ClayAlertBase extends Component {
 	 * @private
 	 */
 	_resumeTimeout() {
-		if (this._delayTime > 0) {
-			this._startDelayTime = new Date();
+		if (this._timeToDisappear > 0) {
+			this._startedTime = new Date();
 			this._timer = setTimeout(() => {
 				this.close();
-			}, this._delayTime);
+			}, this._timeToDisappear);
+		}
+	}
+
+	/**
+	 * Sets the delayTime if passed, if it does not set the default, and starts.
+	 * @private
+	 */
+	_startTimer() {
+		if (
+			this.autoClose &&
+			(this.type === 'stripe' || this.type === 'toast')
+		) {
+			if (isNumber(this.autoClose)) {
+				this._timeToDisappear = this.autoClose * 1000;
+			} else {
+				this._timeToDisappear = (this.type === 'toast' ? 8 : 5) * 1000;
```

Comment:
We should not have here the defaults but in the specific components (ClayToast and ClayStripe). Due in those components `autoClose` can be `bool` or `number` the default should be specified in its templates.

Reply:
hey @carloslancha I've been thinking of changing the `autoClose` approach to `bool` or `number` in the components (ClayToast and ClayStripe).

We can make it `Config.oneOf` with the two possive options that Lexicon allows, in the case of Stripe 5 and 10 in Toast 8 and 12. I do not know if it fits in the context of `autoClose` and if it will confuse the API, but we were able to restrict the possibility of developer putting a different value than is allowed.

Disregard the names 🙂
```javascript
autoClose: Config.oneOf(['default', 'action'])
	.setter(value => {
		if (value === 'default') {
			return 8;
		}

		return 12;
	}),
```

What do you think?

cc @jbalsas

Reply:
Hey @matuzalemsteles the idea is to try being less restrictive with this. In this case Lexicon defines that the values should be 8 and 12, but will be the user the one who pass that config to the component, we need to be covered if one team decides that for them the time must be 16.

So here `autoClose` should be `Config.oneOf([Config.bool(), Config.number()]).value(false)`

I like the approach anyway ;)

Thx!

Reply:
That makes sense now, thanks for the clarifications.